cas-server-support-osf/src/test/java/io/cos/cas/mock/MockNormalizeRemotePrincipal.java

```diff
@@ -0,0 +1,38 @@
+package io.cos.cas.mock;
+
+import io.cos.cas.authentication.OpenScienceFrameworkCredential;
+
+import org.jasig.cas.CentralAuthenticationService;
+
+import org.json.JSONObject;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+/**
+ * This class mocks the {@link OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction} class.
+ *
+ * @author Longze Chen
+ * @since  4.1.5
+ */
+public class MockNormalizeRemotePrincipal extends MockOsfRemoteAuthenticateAction {
+
+    /** Constructor. */
+    public MockNormalizeRemotePrincipal(final CentralAuthenticationService centralAuthenticationService) {
+        super(centralAuthenticationService);
+    }
+
+    @Override
+    protected JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential credential)
```

Comment:
The reason that I do `extends` with `@Override` is that `mockito` cannot mock private/protected methods. The `MockNormalizeRemotePrincipal` is a mock class of `OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction` with the method `normalizeRemotePrincipal` overridden.